### PR TITLE
M3-5462: Keys and secrets modals update UI

### PR DESCRIPTION
**Pull request by codyfinn**
## Description

Adds the ability to download the keys for a token from the text field that displays it.

## How to test

1. Navigate to /profile/tokens
2. Create a new Personal Access Token
3. Ensure that the `TextFields` displaying the keys have both a download and copy `Tooltip` at the end of the `TextField`
4. Click the download `Tooltip`
5. Ensure the file downloaded
    - Contains the string displayed in the `TextField`
    - Is named the label of the `TextField` in snake case

### CODE REVIEW

**Review by HanaXu — COMMENTED** [review]
We'll also need to check the Object Storage Access Keys modal (see below)

The tooltip for the download button says Copy instead of Download and the order is reverse of the mockup
Screen Shot 2022-06-24 at 3 19 41 PM

And for the Object Storage Access Keys modal specifically, would it make more sense to add a secondary download button at the bottom of the modal to download both keys at once instead of adding separate download buttons?

File: packages/manager/src/components/CopyableAndDownloadableTextField.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  props
+) => {
+  const classes = useStyles();
+  const { value, className, hideIcon, ...restProps } = props;
+
+  return (
+    <TextField
+      value={value}
+      {...restProps}
+      className={`${className} ${'copy'} ${classes.removeDisabledStyles}`}
```
Comment (DevDW) [inline packages/manager/src/components/CopyableAndDownloadableTextField.tsx:42]: I think we can remove the `${' ... '}` wrapping `copy` here

File: packages/manager/src/features/Profile/SecretTokenDialog/SecretTokenDialog.tsx [hunk 2]
```diff
@@ -27,42 +29,64 @@ const useStyles = makeStyles(() => ({
...
 
+const renderActions = (onClose: () => void) => (
+  <ActionsPanel>
+    <Button
+      buttonType="secondary"
+      onClick={onClose}
+      data-qa-cancel
+      data-testid="dialog-cancel"
+    >
+      Cancel
```
Comment (DevDW) [inline packages/manager/src/features/Profile/SecretTokenDialog/SecretTokenDialog.tsx:40]: I mistakenly referred to it as a Cancel button in my last comment, but the mockups show "Close".

That brings up a worthwhile point, though – clicking either action button results in the modal closing. Maybe we should just have the "I Have Saved My XYZ" button? @matthewbartholomew / @mbartholomew-design

File: packages/manager/src/features/Profile/SecretTokenDialog/SecretTokenDialog.tsx [hunk 3]
```diff
@@ -27,42 +29,64 @@ const useStyles = makeStyles(() => ({
...
       maxWidth="sm"
+      actions={actions}
     >
       <Notice
         spacingTop={8}
         warning
         className={classes.noticeText}
-        text={`For security purposes, we can only display your ${
+        text={`${
+          objectStorageKey ? '' : 'Your keys have been generated.'
```
Comment (DevDW) [inline packages/manager/src/features/Profile/SecretTokenDialog/SecretTokenDialog.tsx:73]: I thought `Your keys have been generated.` should only be added for the Access Keys modal?

File: packages/manager/src/features/Profile/SecretTokenDialog/SecretTokenDialog.tsx (hunk 2 above)
Comment (HanaXu) [reply packages/manager/src/features/Profile/SecretTokenDialog/SecretTokenDialog.tsx:40]: Agreed

**Review by HanaXu — COMMENTED** [review]
Can we include the label in the generated file name? That would make it easier to differentiate between different token files

**Comment by codyfinn** [comment]
> Can we include the label in the generated file name? That would make it easier to differentiate between different token files

It is definitely possible, but I think for the first pass this works well enough. How we should handle naming the file is something I think the team should discuss. Due to how a user may have labeled their bucket or token how we format the rest of the file name could make it look weird or be hard to find.

File: packages/manager/src/components/DownloadTooltip.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
+import classNames from 'classnames';
+import { downloadFile } from 'src/utilities/downloadFile';
+import * as React from 'react';
+import FileDownload from 'src/assets/icons/download.svg';
```
Comment (bnussman) [inline packages/manager/src/components/DownloadTooltip.tsx:4]: The download icon looks a bit to small and maybe a different color than the copy icon? 

Screen Shot 2022-07-07 at 12 39 06 PM

File: packages/manager/src/components/CopyableAndDownloadableTextField.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import _ from 'lodash';
```
Comment (bnussman) [inline packages/manager/src/components/CopyableAndDownloadableTextField.tsx:2]:
```suggestion
```

Remove this `import`

**Review by jdamore-linode — APPROVED** [review]

**Pull request merged**